nservicebus/azure-service-bus/performance-tuning_body_asb_[7,).partial.md

```diff
@@ -0,0 +1,63 @@
+
+Slow:
+
+Snippet: asb-slow-send
+
+Fast:
+
+Snippet: asb-fast-send
+
+## Tuning receive performance
+
+Tuning receive performance depends on quite a few things that vary per scenario, but the following settings bring improvements for all scenarios.
+
+  * `MessagingFactories().NumberOfMessagingFactoriesPerNamespace()`: Each factory manages a TCP connection to a front end node in the broker, this connection has throughput limits. Opening more factories significantly improves receive performance. Optimal value ares between 16 and 40 for receiving.
+  * `NumberOfClientsPerEntity()`: Keep this value equal to the number of factories, this ensures there is 1 internal receive client per factory.
+
+Other settings that impact performance a lot, but their values differ quite a bit per scenario, are:
+
+  * `LimitMessageProcessingConcurrencyTo()`: This is the global concurrency limit across all receive entities. Ensure this value matches the per receiver desired concurrency for the scenario, multiplied by the number of receivers.
+  * `MessageReceivers().PrefetchCount()`: Value is per receiver, in general the following rule applies: The lower the throughput rate of an individual receiver, the lower this value should be.
+  * `Transactions()`: The most significant performance impact comes from the transactional guarantees needed, `SendsAtomicWithReceive` will offer less performance then `ReceiveOnly`.
+
+### Sends Atomic With Receive
+
+In order to guarantee that sends occur as an atomic operation with the receive confirmation, the ASB SDK requires that both operations are enclosed in a `Serializable` transaction. This implies that:
+
+  * All atomic operations are executed in a 'Serial' fashion, i.e. 1 by 1, therefore it does not make sense to allow many concurrent operations per client, nor does it make much sense to prefetch a lot of messages.
+	  - Concurrency: Optimal values are 2 or 4 per client
+	  - PrefetchCount: Optimal value is 20
+  * Setting up the transactional connection takes up quite some time, this causes the first operational minutes to perform very 'bumpy', but after a while the throughput becomes consistent.
+  * All send operations that occur in the scope of a single receive (inside a handler), must fit into the transaction. Therefore a maximum of 100 messages can be sent this way. The good news is that doing so has little additional implications on performance tuning as 1 send operation or 100 is actually the same thing except for the size of the operation.
+
+NOTE: Send operations inside a handler are only executed when the handler operation returns, unless the send options request for immediate dispatch. Therefore using `await` on a Send inside a handler will probably not seem to impact performance negatively, this is because no send is actually occurring, yet it is still advised to return the task instead of awaiting.
```

Comment:
All outgoing operations are batched by default unless explicitly requested otherwise. Should we just link to batched dispatch here as well as to the async handler documentation? Therefore remove the note?